src/blocks/scratch3_event.js

```diff
@@ -25,7 +25,8 @@ Scratch3EventBlocks.prototype.getHats = function () {
             restartExistingThreads: true
         },
         'event_whenkeypressed': {
-            restartExistingThreads: false
+            restartExistingThreads: false,
+            edgeActivated: true
```

Comment:
But this shouldn't be an edge-activated hat - there's no need to check the predicate on every frame.